### PR TITLE
Misp - added data filtering

### PR DESCRIPTION
**Pull request by teizenman**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/20277

## Description
Most of the data is stored in `Attributes` section which exists for every event and every related event.

### CODE REVIEW

File: Integrations/MISP_V2/MISP_V2.py [hunk 1]
```diff
@@ -349,7 +369,13 @@ def get_dbot_level(threat_level_id: str) -> int:
     return 0
 
 
-def check_file():
+def get_files_events():
+    files = argToList(demisto.args().get('file'), ',')
+    for file in files:
```
Comment (ronykoz) [inline Integrations/MISP_V2/MISP_V2.py]: File is a python saved word, use something else please

File: Integrations/MISP_V2/MISP_V2.py [hunk 2]
```diff
@@ -231,6 +232,24 @@ def replace_keys(obj_to_build: Union[dict, list, str]) -> Union[dict, list, str]
...
+
+    # each event has it's own attributes
+    for attribute in context_data[0]['Attribute']:
+        for key in list(attribute.keys()):
+            if key not in DATA_KEYS_TO_SAVE:
+                del attribute[key]
+
+    # each related event has it's own attributes
+    for obj in context_data[0]['Object']:
+        for attribute in obj['Attribute']:
```
Comment (ronykoz) [inline Integrations/MISP_V2/MISP_V2.py]: Seems like code duplication, separate to another function

File: Integrations/MISP_V2/MISP_V2.yml [hunk 3]
```diff
@@ -19,6 +19,29 @@ configuration:
   name: insecure
   required: false
   type: 8
+- display: Select which event's attribute data fields will be saved to context. Default
```
Comment (ronykoz) [inline Integrations/MISP_V2/MISP_V2.yml]: Not clear for how the user will choose “all” maybe mention that by leaving it empty it will take all the fields

**Comment by teizenman** [comment]
@kirbles19 Please verify RN and arguments at:
1. https://github.com/demisto/content/blob/misp/Integrations/MISP_V2/CHANGELOG.md
2. https://github.com/demisto/content/blob/misp/Integrations/MISP_V2/MISP_V2.yml

**Comment by ronykoz** [comment]
@teizenman maybe you should add a test for this new logic, just to verify it will not break in the future

File: Integrations/MISP_V2/test_constants.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
+full_response_before_filtering = [
```
Comment (ronykoz) [inline Integrations/MISP_V2/test_constants.py]: We shouldn’t have another py file in the integration directory, if you want to add files that are relevant to the tests add a directory called test_data and put those files in there

**Comment by ronykoz** [comment]
You should fix the import in the test as well for this to work

**Comment by teizenman** [comment]
It finds the import, the UT passed successfully both locally and in circle.

**Comment by kirbles19** [comment]
@teizenman Done.

**Pull request merged**